### PR TITLE
Add support for get_enqueued_local_size()

### PR DESCRIPTION
**Pull request by kpet**
Start implementing support for non-uniform NDRanges, following the approach outlined in #522 

Signed-off-by: Kévin Petit <kpet@free.fr>

### CODE REVIEW

**Comment by kpet** [comment]
I should have explained this in my commit message. Vulkan doesn't support non-uniform workgroup sizes at all at the moment so we will need some kind of emulation until it does. As discussed in our last F2F sync, I'm planning to add support to clvk to split non-uniform NDRanges into multiple dispatches as a first step. With that approach clspv would only have to know about/support uniform regions.

**Comment by alan-baker** [comment]
Can you file an issue with a summary of how you intend to non-uniform workgroups?

**Comment by kpet** [comment]
I've described the overall approach in https://github.com/google/clspv/issues/522.

File: lib/DeclarePushConstantsPass.cpp [hunk 1]
```diff
@@ -62,6 +62,13 @@ bool DeclarePushConstantsPass::runOnModule(Module &M) {
     PushConstants.emplace_back(clspv::PushConstant::GlobalOffset);
   }
 
+  if ((clspv::Option::Language() ==
+       clspv::Option::SourceLanguage::OpenCL_C_20) ||
+      (clspv::Option::Language() ==
+       clspv::Option::SourceLanguage::OpenCL_CPP)) {
```
Comment (alan-baker) [inline lib/DeclarePushConstantsPass.cpp]: This should also check if get_enqueued_local_size is used. Seems unnecessary to unconditionally require this value.

Comment (kpet) [reply lib/DeclarePushConstantsPass.cpp]: Yes, agree. This is a problem for `get_work_dim()` as well and I thought I'd tackle that as a separate PR but I don't mind doing it here if that's the preference.

Comment (alan-baker) [reply lib/DeclarePushConstantsPass.cpp]: I'm ok with a separate PR.

Comment (kpet) [reply lib/DeclarePushConstantsPass.cpp]: Ok, I'll make that my next PR then.

**Comment by kpet** [comment]
I've updated the commit message, PTAL.

**Pull request merged**